tasks/configure.yml

```diff
@@ -12,15 +12,15 @@
   notify:
     - reload prometheus
 
-- name: copy custom alerting rule files
-  copy:
-    src: "{{ item }}"
+- name: Synchronize custom alerting rule files
+  synchronize:
+    src: "{{ item | dirname }}/"
     dest: "{{ prometheus_config_dir }}/rules/"
-    owner: root
-    group: prometheus
-    mode: 0640
-    validate: "{{ _prometheus_binary_install_dir }}/promtool check rules %s"
-  with_fileglob: "{{ prometheus_alert_rules_files }}"
+    delete: true
+    perms: false
+    group: false
+    owner: false
+  loop: "{{ prometheus_alert_rules_files }}"
```

Comment:
By using this `loop` source and `{{ item | dirname }}`, does it mean that we'll run synchronize multiple times for the same directory?

E.g.

```text
- rules-dev/file-a.rule
- rules-dev/file-b.rule
- rules-dev/file-c.rule
```

In this case, we'll `synchronize` `rules-dev` thrice? We can improve that later, but do you expect any issues with this behavior?

Reply:
Good catch but not for now. Here is our config:
```prometheus_alert_rules_files:
  - "files/prometheus/rules/{{ env }}/*.rules"
```

So the list only have one entry.